ui/src/app/components/app-router.tsx

```diff
@@ -59,3 +59,19 @@ export const Navigate = ({to}): React.ReactElement => {
   const location = useLocation();
   return <Redirect to={{pathname: to, state: {from: location}}}/>;
 };
+
+// This is essentially a setState for navigation.
+// When the navigation state is set to true it will trigger a render.
+// At this point the state should be used to render the 'Navigate' component, triggering a route change
+// This hook will automatically set the state back to false, preventing the need for an additional 'useEffect' per nav item
+export const useNavigationState = (): [boolean, Function] => {
```

Comment:
There's really nothing navigation-specific here. Could  this be more of a `useToggle`?

Reply:
I like that a lot better - thanks!